src/rgw/rgw_admin.cc

```diff
@@ -9139,11 +9139,11 @@ int main(int argc, const char **argv)
       }
     }
 
-    pipe->source.add_zones(*opt_source_zone_ids);
+    pipe->source.add_zones(opt_source_zone_ids);
```

Comment:
instead of changing `add_zones()` to take an optional vector, please just make these calls conditional:
```c++
    if (opt_source_zone_ids) {
      pipe->source.add_zones(*opt_source_zone_ids);
```

Reply:
ok, update